### ISSUE TITLE
Can't locate isrcore/Shell.pm in @INC

### ISSUE THREAD

**Issue by celtare21** [infobyte/evilgrade#16]
Hello guys! Im trying to run evilgrade but i always get an error, i tried changing os'es, using git clone or aptitude but nothing works. This is the error:
"Can't locate isrcore/Shell.pm in @INC (you may need to install the isrcore::Shell module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at ./evilgrade line 25.
BEGIN failed--compilation aborted at ./evilgrade line 25."

**Comment by liuta** [comment]
this is what i get when i try to run evilgrade.
Can't locate isrcore/Shell.pm in @INC (you may need to install the isrcore::Shell module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at ./evilgrade line 25.
BEGIN failed--compilation aborted at ./evilgrade line 25.
solution please, from anyone thanks

**Comment by liuta** [comment]
Can't locate isrcore/Shell.pm in @INC (you may need to install the isrcore::Shell module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at ./evilgrade line 25.
BEGIN failed--compilation aborted at ./evilgrade line 25.
please if you can,email the solution to me on jett.play@yahoo.com
i will be grateful.

**Comment by pocoZ** [comment]
Короче, нужно всю директорию '/isrcore/' (полный путь у меня '/root/evilgrade/isrcore/') перенести в '/etc/perl/'
In short, you need the entire directory /isrcore/ (full path I '/root/evilgrade/isrcore/') moved to the '/etc /perl/'
Как же я задолбался пока понял в чем дело)

**Comment by Snipercatz** [comment]
I assume this problem hasn't been solved yet since the issue is still open and the tip provided by pocoZ does not work, for me at least.  I am getting the exact same response error as has been listed by at least 10 other people in previous threads.  I am not using a VM.  I have the latest version of Kali Linux 2.0 installed as my operating system.  I've now wasted the past hour of my life trying to dick with this.  giving up at this point.  It'd really be nice to use the tool though so I guess I'll wait until there's a fix.  #irritated

**Comment by f-amato** [comment]
This is a duplicate issue #11 already fixed in the last edition. You try updated using git pull or do a git clone from github.
The Kali apt it is not updated

**Comment by sbrun** [comment]
Hi,

I just packaged for Kali the latest version 2.0.8. 

FYI I had to patch the code to make it worked (I had the same error). I think the issue was because of Perl: since Perl version 5.22.2-5, '.' has been removed from @INC by default in Debian so in Kali too.
I added these 2 lines at the beginning of file evilgrade:
`use FindBin;`
`use lib $FindBin::Bin;`

**Comment by f-amato** [comment]
Thanks @sbrun !
Best

**Comment by Snipercatz** [comment]
hey @sbrun damn...I still can't get it....
Here's what my file now looks like:
#!/usr/bin/perl
###############
# evilgrade
#
# Copyright 2010 Francisco Amato
#
# This file is part of isr-evilgrade, www.infobytesec.com .
#
# isr-evilgrade is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation version 2 of the License.
#
# isr-evilgrade is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with isr-evilgrade; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#
# '''
##
use FindBin;
use lib $FindBin::Bin;
use isrcore::Shell;
use isrcore::shellz;

&main;

---snip.
Did I place it incorrectly?

Here's my error: 
root@kali:~/Downloads/evilgrade# perl evilgrade 
Can't locate RPC/XML.pm in @INC (you may need to install the RPC::XML module) (@INC contains: /root/Downloads/evilgrade /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /root/Downloads/evilgrade/isrcore/shellz.pm line 36.
Compilation failed in require at evilgrade line 27.
BEGIN failed--compilation aborted at evilgrade line 27.

**Comment by pocoZ** [comment]
in your terminal:
`cpan RPC::XML`
do also here it should be - Data::Dump, Digest::MD5, Time::HiRes

**Comment by Snipercatz** [comment]
F YEAH IT WORKED!!! xoxoxoxoxox

Thanks guys. Sorry for the 'tude in the beginning!

**Comment by ptomar25** [comment]
Give a try to this. It worked for me.
Copy isrcore folder from evilgrade to /etc/perl/ and then try running evilgrade. Hope it works !!!!

**Comment by liuta** [comment]
please brother,can you please help me with a site to buy good and working cards?And also a good bin to use in buying the card.                                                                                         thanks for been there 

    On Sunday, November 13, 2016 4:35 AM, pocoZ <notifications@github.com> wrote:
 

 Короче, нужно всю директорию '/isrcore/' (полный путь у меня '/root/evilgrade/isrcore/') перенести в '/etc/perl/'
In short, you need the entire directory /isrcore/ (full path I '/root/evilgrade/isrcore/') moved to the '/etc /perl/'
Как же я задолбался пока понял в чем дело)—

**Comment by pocoZ** [comment]
I do not understand exactly what you need. Write details in the email felixmayakovsky@gmail.com

**Comment by ffaggiani** [comment]
@ptomar25  

> Copy isrcore folder from evilgrade to /etc/perl/ and then try running evilgrade. Hope it works !!!!

With this and deps installed , its working like a charm

**Comment by Rolence** [comment]
thanks it works for me but i have to run the service using ./evilgrade every time i want to use it. Is not not possible to include evilgrade in the root path of the system?

**Comment by kanken57** [comment]
I am really pissed right now,-  I have just encountered same problem and same fix in Jan. 2019.
Why haven't this been fixed yet?? Too bad!!
